Sources/HAP/Server/Server.swift

```diff
@@ -38,8 +38,13 @@ public class Server: NSObject, NetServiceDelegate {
         device.controllerHandler!.removeSubscriptions = device.removeSubscriberForAllCharacteristics
 
         let applicationHandler = ApplicationHandler(responder: root(device: device))
-
-        group = MultiThreadedEventLoopGroup(numberOfThreads: numberOfThreads)
+        
+        if let worker = worker {
+            self.group = worker
+        } else {
+            self.group = MultiThreadedEventLoopGroup(numberOfThreads: System.coreCount)
+        }
```

Comment:
I haven't written Swift for some time, but shouldn't this also work?
```suggestion
        self.group = worker ?? MultiThreadedEventLoopGroup(numberOfThreads: System.coreCount)
```